### PR TITLE
"align_storage" only overrides strides instead of both extents and strides

### PR DESCRIPTION
**Pull request by jingpu**
This helps to avoid over-computation in the extern stage when using
the "align_storage" directive.

### CODE REVIEW

File: src/StorageFlattening.cpp [hunk 1]
```diff
@@ -118,6 +118,11 @@ class FlattenDimensions : public IRMutator {
             shader_scope_realizations.pop(op->name);
         }
 
+        // The allocation bounds of the function taken into account of
+        // the align_storage directives. It is only used to determine the
+        // host allocation size and the strides in buffer_t objects (which
+        // also affects the device allocation in some backends).
+        vector<Expr> allocation_bounds(extents.size());
```
Comment (dsharletg) [inline src/StorageFlattening.cpp]: allocation_bounds -> allocation_extents? Generally, "bounds" refers to both a min and an extent.

**Comment by abadams** [comment]
lgtm

**Comment by abadams** [comment]
Actually this probably needs a test to verify we don't break it in future. E.g. maybe add an align_storage directive to the extern_producer test and an assertion in the extern stage that the strides and extents are as expected.

**Comment by abadams** [comment]
Test failure is a timeout.

**Comment by abadams** [comment]
This seems like the sort of thing that could have obscure consequences, so I'll pull it into the repo and open a fresh PR to get the buildbots to run

**Comment by abadams** [comment]
Moved to #2436

**Pull request closed without merging**